interlok-rest-metrics-jvm/src/main/java/com/adaptris/rest/metrics/jvm/JvmMetricsGenerator.java

```diff
@@ -39,123 +54,16 @@ public JvmMetricsGenerator(Collection<Tag> tags) {
-          .baseUnit(BaseUnits.BYTES).register(registry);
-    }
-    
-    ThreadMXBean threadBean = ManagementFactory.getThreadMXBean();
-
-    Gauge.builder("jvm.threads.peak", threadBean, ThreadMXBean::getPeakThreadCount)
-            .tags(tags)
-            .description("The peak live thread count since the Java virtual machine started or peak was reset")
-            .baseUnit(BaseUnits.THREADS)
-            .register(registry);
-
-    Gauge.builder("jvm.threads.daemon", threadBean, ThreadMXBean::getDaemonThreadCount)
-            .tags(tags)
-            .description("The current number of live daemon threads")
-            .baseUnit(BaseUnits.THREADS)
-            .register(registry);
-
-    Gauge.builder("jvm.threads.live", threadBean, ThreadMXBean::getThreadCount)
-            .tags(tags)
-            .description("The current number of live threads including both daemon and non-daemon threads")
-            .baseUnit(BaseUnits.THREADS)
-            .register(registry);
-
-    try {
-        threadBean.getAllThreadIds();
-        for (Thread.State state : Thread.State.values()) {
-            Gauge.builder("jvm.threads.states", threadBean, (bean) -> getThreadStateCount(bean, state))
-                    .tags(Tags.concat(tags, "state", getStateTagValue(state)))
-                    .description("The current number of threads having " + state + " state")
-                    .baseUnit(BaseUnits.THREADS)
-                    .register(registry);
-        }
-    } catch (Error error) {
-        // An error will be thrown for unsupported operations
-        // e.g. SubstrateVM does not support getAllThreadIds
-    }
-    
-    ClassLoadingMXBean classLoadingBean = ManagementFactory.getClassLoadingMXBean();
-
-    Gauge.builder("jvm.classes.loaded", classLoadingBean, ClassLoadingMXBean::getLoadedClassCount)
-            .tags(tags)
-            .description("The number of classes that are currently loaded in the Java virtual machine")
-            .baseUnit(BaseUnits.CLASSES)
-            .register(registry);
-
-    FunctionCounter.builder("jvm.classes.unloaded", classLoadingBean, ClassLoadingMXBean::getUnloadedClassCount)
-            .tags(tags)
-            .description("The total number of classes unloaded since the Java virtual machine has started execution")
-            .baseUnit(BaseUnits.CLASSES)
-            .register(registry);
-    
-    CompilationMXBean compilationBean = ManagementFactory.getCompilationMXBean();
-    if (compilationBean != null && compilationBean.isCompilationTimeMonitoringSupported()) {
-        FunctionCounter.builder("jvm.compilation.time", compilationBean, CompilationMXBean::getTotalCompilationTime)
-                .tags(Tags.concat(tags, "compiler", compilationBean.getName()))
-                .description("The approximate accumulated elapsed time spent in compilation")
-                .baseUnit(BaseUnits.MILLISECONDS)
-                .register(registry);
-    }
+    meterBinders.forEach( binder -> {
```

Comment:
I wonder if it's worth doing something like this instead?

```
  private final List<Supplier<MeterBinder>> meterBinders = Arrays.asList(
      () -> new ClassLoaderMetrics(getTags()), etc etc etc
  );

  @Override
  public void bindTo(MeterRegistry registry) {
    meterBinders.forEach( binder -> {
      binder.get().bindTo(registry);
    });
  }
```

Then you don't get the missing coverage around the catch Exception; since you're using the no-args constructor, but we can see that Meters can take an Iterable<Tag>...

Reply:
This is true however I don't think we want to add our tags to the JVM stats.
We'll see though after Matt's had a go..